app/components/custom_viewer_component.rb

```diff
@@ -5,12 +5,19 @@
 class CustomViewerComponent < Blacklight::Component
   attr_reader :document, :presenter, :view_config, :classes, :block_context
 
+  SirTrevorBlock = Struct.new(:maxheight, :item)
+
   def initialize(document:, presenter:, view_config: nil, block_context: nil, **kwargs)
     super
 
     @document = document
     @presenter = presenter
     @view_config = view_config
-    @block_context = block_context
+    @block_context = correct_block(block_context)
+  end
+
+  def correct_block(block_context)
```

Comment:
The name `correct_block` seems vague. It seems to be selecting/creating a block based on the document id, so maybe `block_for_document` or something?